packages/itwin/grouping-mapping-widget/src/widget/components/CalculatedPropertyActionWithVisuals.tsx

```diff
@@ -57,6 +60,19 @@ export const CalculatedPropertyActionWithVisuals = ({
   const [validator, showValidationMessage] = useValidator();
   const [resolvedHiliteIds, setResolvedHiliteIds] = useState<string[]>([]);
   const [colorProperty, setColorProperty] = useState<boolean>(false);
+  const origPropertyName = calculatedProperty?.propertyName ?? "";
+  const [showSaveValidationModal, setShowSaveValidationModal] = useState<boolean>(false);
+  const [invalidCustomCalcs, setInvalidCustomCalcs] = useState<invalids[]>([]);
+
+  async function checkOutliers(changedPropName: string): Promise<invalids[]> {
+    const accessToken = await getAccessToken();
+    const [customCalcProps] = await Promise.all([
+      mappingClient.getCustomCalculations(accessToken, iModelId, mappingId, group.id),
+    ]);
+    const changes: invalids[] = await PropertyValidation({origPropertyName, changedPropertyName: changedPropName, customCalcProps});
+    setInvalidCustomCalcs(changes);
+    return Promise.resolve(changes);
+  }
```

Comment:
`checkOutliers` looks like it can be made more generic. I see a lot of duplicated code.

Reply:
I tried a few things to make it more generic and include all the other calls in the "PropertyValidationUtils" but due to the structuring of the files and functions, it is getting complicated due to the other hooks.